lib/services/amq_event_service.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Aprb.Service.AmqEventService do
+  use GenServer
+  use AMQP
+
+  alias Aprb.Service.EventService
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, [])
+  end
+
+  def init(topic) do
+    rabbitmq_connect(topic)
+  end
+
+  defp rabbitmq_connect(topic) do
+    case Connection.open(Application.get_env(:aprb, RabbitMQ)) do
+      {:ok, conn} ->
+        # Get notifications when the connection goes down
+        Process.monitor(conn.pid)
+        {:ok, chan} = Channel.open(conn)
+        Basic.qos(chan, prefetch_count: 10)
+        Exchange.topic(chan, topic, durable: true)
+        queue_name = "aprb_#{topic}_queue"
```

Comment:
Nice!  You can have a consumer bound to different queues on a per-topic basis or create multiple bindings on one queue and receive all messages there.  However, in the latter case your application will need to look at the message's `routing_key` property to handle which topic/exchange it came from.  This is the most straightforward solution :)